src/indexing/test/indexing.cpp

```diff
@@ -614,10 +617,14 @@ TEST_CASE(
 
         handled_writes = writes.end() - writes.begin();
 
-        if (loops++ > 100)
+        if (work_done)
         {
-          throw std::logic_error("Looks like a permanent loop");
+          if (post_work_done_loops++ > 100)
```

Comment:
100 seems arbitrary?

Reply:
The 100 is extremely arbitrary! We expect this thread to provide all of the entries the other threads are looking for extremely quickly (ie - in <5 loops), so 100 is aiming to be a "definite failure". We could probably safely make it a little lower, but we wouldn't noticeably change execution time (even time-to-failure), and risk getting false negatives - failures where we claim a permanent loop when the system is actually making progress.